pkg/scheduler/framework/interface.go

```diff
@@ -145,6 +145,48 @@ func NewPodsToActivate() *PodsToActivate {
 	return &PodsToActivate{Map: make(map[string]*v1.Pod)}
 }
 
+// StatusError is an implementation of error.
+type StatusError struct {
+	message string
+	err     error
+}
+
+// Error returns the error message of StatusError.
+func (e *StatusError) Error() string {
+	if e.err != nil {
+		return e.err.Error()
+	}
+	return e.message
+}
+
+// Is returns if the error message matches the target.
+func (e *StatusError) Is(target error) bool {
+
+	if sErr, ok := target.(*StatusError); ok {
+		return e.Error() == sErr.Error()
+	}
+
+	if e.err != nil {
+		return errors.Is(e.err, target)
+	}
```

Comment:
```suggestion
	if e.err != nil {
		return errors.Is(err, target)
	}
	if sErr, ok := target.(*StatusError); ok {
		return e.message == sErr.Message
	}
```

In particular, we should avoid comparing `e.Error()`. It gives up the types.

Reply:
I made all the suggested changes and run the unit tests but some of them failed. Do you have ideas what to change?
```--- FAIL: TestPlugin (0.00s)
    --- FAIL: TestPlugin/delayed-allocation-scheduling-finish-concurrent-label-update (0.11s)
        dynamicresources.go:323: INFO pod resource claims pod="default/my-pod" resourceclaims=[default/my-pod-my-resource]
        dynamicresources.go:579: INFO all potential nodes already set pod="default/my-pod" potentialnodes=[&Node{ObjectMeta:{worker      0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[nodename:worker] map[] [] [] []},Spec:NodeSpec{PodCIDR:,DoNotUseExternalID:,ProviderID:,Unschedulable:false,Taints:[]Taint{},ConfigSource:nil,PodCIDRs:[],},Status:NodeStatus{Capacity:ResourceList{pods: {{32 0} {<nil>} 32 DecimalSI},},Allocatable:ResourceList{pods: {{32 0} {<nil>} 32 DecimalSI},},Phase:,Conditions:[]NodeCondition{},Addresses:[]NodeAddress{},DaemonEndpoints:NodeDaemonEndpoints{KubeletEndpoint:DaemonEndpoint{Port:0,},},NodeInfo:NodeSystemInfo{MachineID:,SystemUUID:,BootID:,KernelVersion:,OSImage:,ContainerRuntimeVersion:,KubeletVersion:,KubeProxyVersion:,OperatingSystem:,Architecture:,},Images:[]ContainerImage{},VolumesInUse:[],VolumesAttached:[]AttachedVolume{},Config:nil,},}]
        dynamicresources_test.go:605: Updating *v1alpha1.PodScheduling "my-pod", diff (-old, +new):
              &v1alpha1.PodScheduling{
                TypeMeta: {},
                ObjectMeta: v1.ObjectMeta{
                        ... // 8 identical fields
                        DeletionTimestamp:          nil,
                        DeletionGracePeriodSeconds: nil,
            -           Labels:                     nil,
            +           Labels:                     map[string]string{"hello": "world"},
                        Annotations:                nil,
                        OwnerReferences:            {{APIVersion: "v1", Kind: "Pod", Name: "my-pod", UID: "1234", ...}},
                        ... // 2 identical fields
                },
                Spec:   {PotentialNodes: {"worker"}},
                Status: {ResourceClaims: {{Name: "my-resource"}, {Name: "my-resource-2"}}},
              }
        dynamicresources.go:686: INFO start allocation pod="default/my-pod" node="worker"
        dynamicresources.go:174: INFO Updating PodScheduling podscheduling="default/my-pod"
        dynamicresources.go:688: ERROR dynamic resource plugin failed err="ResourceVersion must match the object that gets updated"
        --- FAIL: TestPlugin/delayed-allocation-scheduling-finish-concurrent-label-update/reserve (0.00s)
            dynamicresources_test.go:500: Unexpected status (-want, +got):
                  (*framework.Status)(
                -       &⟪0xc0002fb000⟫{
                -               code: 1,
                -               err:  &⟪0xc0003f76f0⟫"errors".errorString{s: "ResourceVersion must match the object that gets updated"},
                -       },
                +       &⟪0xc0007dd7c0⟫{
                +               code: 1,
                +               err:  &⟪0xc000434050⟫"errors".errorString{s: "ResourceVersion must match the object that gets updated"},
                +       },
                  )
    --- FAIL: TestPlugin/delayed-allocation-missing-class (0.10s)
        dynamicresources.go:323: INFO pod resource claims pod="default/my-pod" resourceclaims=[default/my-pod-my-resource]
        dynamicresources.go:425: ERROR dynamic resource plugin failed err="look up resource class: resourceclass.resource.k8s.io \"my-resource-class\" not found"
        --- FAIL: TestPlugin/delayed-allocation-missing-class/filter/worker (0.00s)
            dynamicresources_test.go:470: Unexpected status (-want, +got):
                  (*framework.Status)(
                -       &⟪0xc0002faec0⟫{
                -               code: 1,
                -               err: &⟪0xc0003f74f0⟫"errors".errorString{
                -                       s: `look up resource class: resourceclass.resource.k8s.io "my-resource-class" not found`,
                -               },
                -       },
                +       &⟪0xc0008fffc0⟫{
                +               code: 1,
                +               err: &⟪0xc000a1ddc0⟫"errors".errorString{
                +                       s: `look up resource class: resourceclass.resource.k8s.io "my-resource-class" not found`,
                +               },
                +       },
                  )
FAIL
FAIL    k8s.io/kubernetes/pkg/scheduler/framework/plugins/dynamicresources      2.668s
?       k8s.io/kubernetes/pkg/scheduler/framework/plugins/examples/multipoint   [no test files]
?       k8s.io/kubernetes/pkg/scheduler/framework/plugins/examples/prebind      [no test files]
?       k8s.io/kubernetes/pkg/scheduler/framework/plugins/examples/stateful     [no test files]
?       k8s.io/kubernetes/pkg/scheduler/framework/plugins/feature       [no test files]
ok      k8s.io/kubernetes/pkg/scheduler/framework/plugins/helper        2.079s
ok      k8s.io/kubernetes/pkg/scheduler/framework/plugins/imagelocality 3.065s
E1206 23:31:22.335597    6375 filtering.go:201] "Node not found"
--- FAIL: TestPreferredAffinity (2.32s)
    --- FAIL: TestPreferredAffinity/invalid_Affinity_fails_PreScore (0.10s)
        scoring_test.go:749: unexpected error: &{1 [] 0xc0003a7d80 }
    --- FAIL: TestPreferredAffinity/invalid_AntiAffinity_fails_PreScore (0.10s)
        scoring_test.go:749: unexpected error: &{1 [] 0xc000160260 }
FAIL
FAIL    k8s.io/kubernetes/pkg/scheduler/framework/plugins/interpodaffinity      13.433s
?       k8s.io/kubernetes/pkg/scheduler/framework/plugins/names [no test files]
ok      k8s.io/kubernetes/pkg/scheduler/framework/plugins/nodeaffinity  3.928s
ok      k8s.io/kubernetes/pkg/scheduler/framework/plugins/nodename      3.333s
ok      k8s.io/kubernetes/pkg/scheduler/framework/plugins/nodeports     4.007s
ok      k8s.io/kubernetes/pkg/scheduler/framework/plugins/noderesources 4.075s
ok      k8s.io/kubernetes/pkg/scheduler/framework/plugins/nodeunschedulable     4.377s
--- FAIL: TestCSILimits (0.00s)
    --- FAIL: TestCSILimits/ephemeral_volume_missing (0.00s)
        csi_test.go:497: Unexpected status (-want, +got):
              (*framework.Status)(
            -   &{
            -           code:    s"Error",
            -           reasons: []string{`looking up PVC test/abc-xyz: persistentvolumeclaim "abc-xyz" not`...},
            -   },
            +   &{
            +           code: s"Error",
            +           err:  e`looking up PVC test/abc-xyz: persistentvolumeclaim "abc-xyz" not found`,
            +   },
              )
    --- FAIL: TestCSILimits/ephemeral_volume_not_owned (0.00s)
        csi_test.go:497: Unexpected status (-want, +got):
              (*framework.Status)(
            -   &{
            -           code:    s"Error",
            -           reasons: []string{"PVC test/abc-xyz was not created for pod test/abc (pod is not ow"...},
            -   },
            +   &{
            +           code: s"Error",
            +           err:  e"PVC test/abc-xyz was not created for pod test/abc (pod is not owner)",
            +   },
              )
--- FAIL: TestEphemeralLimits (0.00s)
    --- FAIL: TestEphemeralLimits/volume_missing (0.00s)
        non_csi_test.go:162: Unexpected status (-want, +got):
              (*framework.Status)(
            -   &{
            -           code:    s"Error",
            -           reasons: []string{`looking up PVC test/abc-xyz: persistentvolumeclaim "abc-xyz" not`...},
            -   },
            +   &{
            +           code: s"Error",
            +           err:  e`looking up PVC test/abc-xyz: persistentvolumeclaim "abc-xyz" not found`,
            +   },
              )
    --- FAIL: TestEphemeralLimits/volume_not_owned (0.00s)
        non_csi_test.go:162: Unexpected status (-want, +got):
              (*framework.Status)(
            -   &{
            -           code:    s"Error",
            -           reasons: []string{"PVC test/abc-xyz was not created for pod test/abc (pod is not ow"...},
            -   },
            +   &{
            +           code: s"Error",
            +           err:  e"PVC test/abc-xyz was not created for pod test/abc (pod is not owner)",
            +   },
              )
E1206 23:31:23.100394    6397 non_csi.go:381] "Unable to parse maximum PD volumes value, using default" err="strconv.Atoi: parsing \"invalid\": invalid syntax"
E1206 23:31:23.100580    6397 non_csi.go:383] "Unable to parse maximum PD volumes value, using default" err="maximum PD volumes is negative"
FAIL
FAIL    k8s.io/kubernetes/pkg/scheduler/framework/plugins/nodevolumelimits      3.817s
ok      k8s.io/kubernetes/pkg/scheduler/framework/plugins/podtopologyspread     7.790s
ok      k8s.io/kubernetes/pkg/scheduler/framework/plugins/queuesort     2.889s
ok      k8s.io/kubernetes/pkg/scheduler/framework/plugins/schedulinggates       3.612s
ok      k8s.io/kubernetes/pkg/scheduler/framework/plugins/selectorspread        7.823s
ok      k8s.io/kubernetes/pkg/scheduler/framework/plugins/tainttoleration       3.968s
?       k8s.io/kubernetes/pkg/scheduler/framework/plugins/testing       [no test files]
ok      k8s.io/kubernetes/pkg/scheduler/framework/plugins/volumebinding 20.383s
?       k8s.io/kubernetes/pkg/scheduler/framework/plugins/volumebinding/metrics [no test files]
ok      k8s.io/kubernetes/pkg/scheduler/framework/plugins/volumerestrictions    4.938s
ok      k8s.io/kubernetes/pkg/scheduler/framework/plugins/volumezone    2.525s
ok      k8s.io/kubernetes/pkg/scheduler/framework/preemption    2.804s
I1206 23:31:29.183132    6469 framework.go:483] "MultiPoint plugin is explicitly re-configured; overriding" plugin="test-plugin"
I1206 23:31:29.183241    6469 framework.go:483] "MultiPoint plugin is explicitly re-configured; overriding" plugin="score-plugin-1"
I1206 23:31:29.183360    6469 framework.go:483] "MultiPoint plugin is explicitly re-configured; overriding" plugin="test-plugin"
I1206 23:31:29.183371    6469 framework.go:483] "MultiPoint plugin is explicitly re-configured; overriding" plugin="score-plugin-1"
I1206 23:31:29.183483    6469 framework.go:483] "MultiPoint plugin is explicitly re-configured; overriding" plugin="test-plugin"
I1206 23:31:29.183492    6469 framework.go:483] "MultiPoint plugin is explicitly re-configured; overriding" plugin="score-plugin-1"
I1206 23:31:29.184113    6469 framework.go:536] "Plugin's EventsToRegister() returned nil" plugin="fakeNoopRuntime"
I1206 23:31:29.184955    6469 framework.go:483] "MultiPoint plugin is explicitly re-configured; overriding" plugin="score-plugin-1"
--- FAIL: TestRunPreFilterPluginsStatus (0.00s)
    framework_test.go:1439: Unexpected status (-want, +got):
          (*framework.Status)(
        -       &{
        -               code: 1,
        -               err: &fmt.wrapError{
        -                       msg: `running PreFilter plugin "prefilter-plugin": injected status`,
        -                       err: &errors.errorString{s: "injected status"},
        -               },
        -               failedPlugin: "prefilter-plugin",
        -       },
        +       &{
        +               code: 1,
        +               err: &fmt.wrapError{
        +                       msg: `running PreFilter plugin "prefilter-plugin": injected status`,
        +                       err: &framework.StatusError{message: "injected status"},
        +               },
        +               failedPlugin: "prefilter-plugin",
        +       },
          )
--- FAIL: TestFilterPlugins (0.00s)
    --- FAIL: TestFilterPlugins/ErrorFilter (0.00s)
        framework_test.go:1619: Unexpected status (-want, +got):
              (*framework.Status)(
            -   &{
            -           code: 1,
            -           err: &fmt.wrapError{
            -                   msg: `running "TestPlugin" filter plugin: injected filter status`,
            -                   err: &errors.errorString{s: "injected filter status"},
            -           },
            -           failedPlugin: "TestPlugin",
            -   },
            +   &{
            +           code: 1,
            +           err: &framework.StatusError{
            +                   err: &fmt.wrapError{
            +                           msg: `running "TestPlugin" filter plugin: injected filter status`,
            +                           err: &framework.StatusError{message: "injected filter status"},
            +                   },
            +           },
            +           failedPlugin: "TestPlugin",
            +   },
              )
        framework_test.go:1622: Unexpected status map (-want, +got):
              framework.PluginToStatus{
            -   "TestPlugin": &{
            -           code: 1,
            -           err: &fmt.wrapError{
            -                   msg: `running "TestPlugin" filter plugin: injected filter status`,
            -                   err: &errors.errorString{s: "injected filter status"},
            -           },
            -           failedPlugin: "TestPlugin",
            -   },
            +   "TestPlugin": &{
            +           code: 1,
            +           err: &fmt.wrapError{
            +                   msg: `running "TestPlugin" filter plugin: injected filter status`,
            +                   err: &framework.StatusError{message: "injected filter status"},
            +           },
            +           failedPlugin: "TestPlugin",
            +   },
              }
    --- FAIL: TestFilterPlugins/ErrorAndErrorFilters (0.00s)
        framework_test.go:1619: Unexpected status (-want, +got):
              (*framework.Status)(
            -   &{
            -           code: 1,
            -           err: &fmt.wrapError{
            -                   msg: `running "TestPlugin1" filter plugin: injected filter status`,
            -                   err: &errors.errorString{s: "injected filter status"},
            -           },
            -           failedPlugin: "TestPlugin1",
            -   },
            +   &{
            +           code: 1,
            +           err: &framework.StatusError{
            +                   err: &fmt.wrapError{
            +                           msg: `running "TestPlugin1" filter plugin: injected filter status`,
            +                           err: &framework.StatusError{message: "injected filter status"},
            +                   },
            +           },
            +           failedPlugin: "TestPlugin1",
            +   },
              )
        framework_test.go:1622: Unexpected status map (-want, +got):
              framework.PluginToStatus{
            -   "TestPlugin1": &{
            -           code: 1,
            -           err: &fmt.wrapError{
            -                   msg: `running "TestPlugin1" filter plugin: injected filter status`,
            -                   err: &errors.errorString{s: "injected filter status"},
            -           },
            -           failedPlugin: "TestPlugin1",
            -   },
            +   "TestPlugin1": &{
            +           code: 1,
            +           err: &fmt.wrapError{
            +                   msg: `running "TestPlugin1" filter plugin: injected filter status`,
            +                   err: &framework.StatusError{message: "injected filter status"},
            +           },
            +           failedPlugin: "TestPlugin1",
            +   },
              }
    --- FAIL: TestFilterPlugins/ErrorAndSuccessFilters (0.00s)
        framework_test.go:1619: Unexpected status (-want, +got):
              (*framework.Status)(
            -   &{
            -           code: 1,
            -           err: &fmt.wrapError{
            -                   msg: `running "TestPlugin1" filter plugin: injected filter status`,
            -                   err: &errors.errorString{s: "injected filter status"},
            -           },
            -           failedPlugin: "TestPlugin1",
            -   },
            +   &{
            +           code: 1,
            +           err: &framework.StatusError{
            +                   err: &fmt.wrapError{
            +                           msg: `running "TestPlugin1" filter plugin: injected filter status`,
            +                           err: &framework.StatusError{message: "injected filter status"},
            +                   },
            +           },
            +           failedPlugin: "TestPlugin1",
            +   },
              )
        framework_test.go:1622: Unexpected status map (-want, +got):
              framework.PluginToStatus{
            -   "TestPlugin1": &{
            -           code: 1,
            -           err: &fmt.wrapError{
            -                   msg: `running "TestPlugin1" filter plugin: injected filter status`,
            -                   err: &errors.errorString{s: "injected filter status"},
            -           },
            -           failedPlugin: "TestPlugin1",
            -   },
            +   "TestPlugin1": &{
            +           code: 1,
            +           err: &fmt.wrapError{
            +                   msg: `running "TestPlugin1" filter plugin: injected filter status`,
            +                   err: &framework.StatusError{message: "injected filter status"},
            +           },
            +           failedPlugin: "TestPlugin1",
            +   },
              }
    --- FAIL: TestFilterPlugins/SuccessAndErrorFilters (0.00s)
        framework_test.go:1619: Unexpected status (-want, +got):
              (*framework.Status)(
            -   &{
            -           code: 1,
            -           err: &fmt.wrapError{
            -                   msg: `running "TestPlugin2" filter plugin: injected filter status`,
            -                   err: &errors.errorString{s: "injected filter status"},
            -           },
            -           failedPlugin: "TestPlugin2",
            -   },
            +   &{
            +           code: 1,
            +           err: &framework.StatusError{
            +                   err: &fmt.wrapError{
            +                           msg: `running "TestPlugin2" filter plugin: injected filter status`,
            +                           err: &framework.StatusError{message: "injected filter status"},
            +                   },
            +           },
            +           failedPlugin: "TestPlugin2",
            +   },
              )
        framework_test.go:1622: Unexpected status map (-want, +got):
              framework.PluginToStatus{
            -   "TestPlugin2": &{
            -           code: 1,
            -           err: &fmt.wrapError{
            -                   msg: `running "TestPlugin2" filter plugin: injected filter status`,
            -                   err: &errors.errorString{s: "injected filter status"},
            -           },
            -           failedPlugin: "TestPlugin2",
            -   },
            +   "TestPlugin2": &{
            +           code: 1,
            +           err: &fmt.wrapError{
            +                   msg: `running "TestPlugin2" filter plugin: injected filter status`,
            +                   err: &framework.StatusError{message: "injected filter status"},
            +           },
            +           failedPlugin: "TestPlugin2",
            +   },
              }
E1206 23:31:29.187683    6469 framework.go:663] "Failed running AddPod on PreFilter plugin" err="injected status" plugin="TestPlugin1" pod=""
--- FAIL: TestFilterPluginsWithNominatedPods (0.00s)
    --- FAIL: TestFilterPluginsWithNominatedPods/node_has_a_high-priority_nominated_pod_and_pre_filters_fail (0.00s)
        framework_test.go:1852: Unexpected status (-want, +got):
              (*framework.Status)(
            -   &{
            -           code: 1,
            -           err: &fmt.wrapError{
            -                   msg: `running AddPod on PreFilter plugin "TestPlugin1": injected status`,
            -                   err: &errors.errorString{s: "injected status"},
            -           },
            -   },
            +   &{
            +           code: 1,
            +           err: &framework.StatusError{
            +                   err: &fmt.wrapError{
            +                           msg: `running AddPod on PreFilter plugin "TestPlugin1": injected status`,
            +                           err: &framework.StatusError{message: "injected status"},
            +                   },
            +           },
            +   },
              )
    --- FAIL: TestFilterPluginsWithNominatedPods/node_has_a_high-priority_nominated_pod_and_filters_fail (0.00s)
        framework_test.go:1852: Unexpected status (-want, +got):
              (*framework.Status)(
            -   &{
            -           code: 1,
            -           err: &fmt.wrapError{
            -                   msg: `running "TestPlugin2" filter plugin: injected filter status`,
            -                   err: &errors.errorString{s: "injected filter status"},
            -           },
            -           failedPlugin: "TestPlugin2",
            -   },
            +   &{
            +           code: 1,
            +           err: &framework.StatusError{
            +                   err: &fmt.wrapError{
            +                           msg: `running "TestPlugin2" filter plugin: injected filter status`,
            +                           err: &framework.StatusError{message: "injected filter status"},
            +                   },
            +           },
            +           failedPlugin: "TestPlugin2",
            +   },
              )
E1206 23:31:29.188167    6469 framework.go:1033] "Failed running PreBind plugin" err="injected status" plugin="TestPlugin" pod="" node=""
E1206 23:31:29.188371    6469 framework.go:1033] "Failed running PreBind plugin" err="injected status" plugin="TestPlugin 1" pod="" node=""
E1206 23:31:29.188525    6469 framework.go:1033] "Failed running PreBind plugin" err="injected status" plugin="TestPlugin" pod="" node=""
E1206 23:31:29.188702    6469 framework.go:1033] "Failed running PreBind plugin" err="injected status" plugin="TestPlugin" pod="" node=""
--- FAIL: TestPreBindPlugins (0.00s)
    --- FAIL: TestPreBindPlugins/ErrorPreBindPlugin (0.00s)
        framework_test.go:2010: Unexpected status (-want, +got):
              (*framework.Status)(
            -   &{
            -           code: 1,
            -           err: &fmt.wrapError{
            -                   msg: `running PreBind plugin "TestPlugin": injected status`,
            -                   err: &errors.errorString{s: "injected status"},
            -           },
            -   },
            +   &{
            +           code: 1,
            +           err: &fmt.wrapError{
            +                   msg: `running PreBind plugin "TestPlugin": injected status`,
            +                   err: &framework.StatusError{message: "injected status"},
            +           },
            +   },
              )
    --- FAIL: TestPreBindPlugins/SuccessErrorPreBindPlugins (0.00s)
        framework_test.go:2010: Unexpected status (-want, +got):
              (*framework.Status)(
            -   &{
            -           code: 1,
            -           err: &fmt.wrapError{
            -                   msg: `running PreBind plugin "TestPlugin 1": injected status`,
            -                   err: &errors.errorString{s: "injected status"},
            -           },
            -   },
            +   &{
            +           code: 1,
            +           err: &fmt.wrapError{
            +                   msg: `running PreBind plugin "TestPlugin 1": injected status`,
            +                   err: &framework.StatusError{message: "injected status"},
            +           },
            +   },
              )
    --- FAIL: TestPreBindPlugins/ErrorSuccessPreBindPlugin (0.00s)
        framework_test.go:2010: Unexpected status (-want, +got):
              (*framework.Status)(
            -   &{
            -           code: 1,
            -           err: &fmt.wrapError{
            -                   msg: `running PreBind plugin "TestPlugin": injected status`,
            -                   err: &errors.errorString{s: "injected status"},
            -           },
            -   },
            +   &{
            +           code: 1,
            +           err: &fmt.wrapError{
            +                   msg: `running PreBind plugin "TestPlugin": injected status`,
            +                   err: &framework.StatusError{message: "injected status"},
            +           },
            +   },
              )
    --- FAIL: TestPreBindPlugins/ErrorAndErrorPlugins (0.00s)
        framework_test.go:2010: Unexpected status (-want, +got):
              (*framework.Status)(
            -   &{
            -           code: 1,
            -           err: &fmt.wrapError{
            -                   msg: `running PreBind plugin "TestPlugin": injected status`,
            -                   err: &errors.errorString{s: "injected status"},
            -           },
            -   },
            +   &{
            +           code: 1,
            +           err: &fmt.wrapError{
            +                   msg: `running PreBind plugin "TestPlugin": injected status`,
            +                   err: &framework.StatusError{message: "injected status"},
            +           },
            +   },
              )
E1206 23:31:29.188979    6469 framework.go:1124] "Failed running Reserve plugin" err="injected status" plugin="TestPlugin" pod=""
E1206 23:31:29.189129    6469 framework.go:1124] "Failed running Reserve plugin" err="injected status" plugin="TestPlugin" pod=""
E1206 23:31:29.189290    6469 framework.go:1124] "Failed running Reserve plugin" err="injected status" plugin="TestPlugin" pod=""
E1206 23:31:29.189480    6469 framework.go:1124] "Failed running Reserve plugin" err="injected status" plugin="TestPlugin" pod=""
E1206 23:31:29.189639    6469 framework.go:1124] "Failed running Reserve plugin" err="injected status" plugin="TestPlugin 1" pod=""
E1206 23:31:29.189794    6469 framework.go:1124] "Failed running Reserve plugin" err="injected status" plugin="TestPlugin" pod=""
E1206 23:31:29.189944    6469 framework.go:1124] "Failed running Reserve plugin" err="injected status" plugin="TestPlugin" pod=""
--- FAIL: TestReservePlugins (0.00s)
    --- FAIL: TestReservePlugins/UnschedulableReservePlugin (0.00s)
        framework_test.go:2168: Unexpected status (-want, +got):
              (*framework.Status)(
            -   &{
            -           code: 1,
            -           err: &fmt.wrapError{
            -                   msg: `running Reserve plugin "TestPlugin": injected status`,
            -                   err: &errors.errorString{s: "injected status"},
            -           },
            -   },
            +   &{
            +           code: 1,
            +           err: &fmt.wrapError{
            +                   msg: `running Reserve plugin "TestPlugin": injected status`,
            +                   err: &framework.StatusError{message: "injected status"},
            +           },
            +   },
              )
    --- FAIL: TestReservePlugins/ErrorReservePlugin (0.00s)
        framework_test.go:2168: Unexpected status (-want, +got):
              (*framework.Status)(
            -   &{
            -           code: 1,
            -           err: &fmt.wrapError{
            -                   msg: `running Reserve plugin "TestPlugin": injected status`,
            -                   err: &errors.errorString{s: "injected status"},
            -           },
            -   },
            +   &{
            +           code: 1,
            +           err: &fmt.wrapError{
            +                   msg: `running Reserve plugin "TestPlugin": injected status`,
            +                   err: &framework.StatusError{message: "injected status"},
            +           },
            +   },
              )
    --- FAIL: TestReservePlugins/UnschedulableReservePlugin#01 (0.00s)
        framework_test.go:2168: Unexpected status (-want, +got):
              (*framework.Status)(
            -   &{
            -           code: 1,
            -           err: &fmt.wrapError{
            -                   msg: `running Reserve plugin "TestPlugin": injected status`,
            -                   err: &errors.errorString{s: "injected status"},
            -           },
            -   },
            +   &{
            +           code: 1,
            +           err: &fmt.wrapError{
            +                   msg: `running Reserve plugin "TestPlugin": injected status`,
            +                   err: &framework.StatusError{message: "injected status"},
            +           },
            +   },
              )
    --- FAIL: TestReservePlugins/ErrorErrorReservePlugins (0.00s)
        framework_test.go:2168: Unexpected status (-want, +got):
              (*framework.Status)(
            -   &{
            -           code: 1,
            -           err: &fmt.wrapError{
            -                   msg: `running Reserve plugin "TestPlugin": injected status`,
            -                   err: &errors.errorString{s: "injected status"},
            -           },
            -   },
            +   &{
            +           code: 1,
            +           err: &fmt.wrapError{
            +                   msg: `running Reserve plugin "TestPlugin": injected status`,
            +                   err: &framework.StatusError{message: "injected status"},
            +           },
            +   },
              )
    --- FAIL: TestReservePlugins/SuccessErrorReservePlugins (0.00s)
        framework_test.go:2168: Unexpected status (-want, +got):
              (*framework.Status)(
            -   &{
            -           code: 1,
            -           err: &fmt.wrapError{
            -                   msg: `running Reserve plugin "TestPlugin 1": injected status`,
            -                   err: &errors.errorString{s: "injected status"},
            -           },
            -   },
            +   &{
            +           code: 1,
            +           err: &fmt.wrapError{
            +                   msg: `running Reserve plugin "TestPlugin 1": injected status`,
            +                   err: &framework.StatusError{message: "injected status"},
            +           },
            +   },
              )
    --- FAIL: TestReservePlugins/ErrorSuccessReservePlugin (0.00s)
        framework_test.go:2168: Unexpected status (-want, +got):
              (*framework.Status)(
            -   &{
            -           code: 1,
            -           err: &fmt.wrapError{
            -                   msg: `running Reserve plugin "TestPlugin": injected status`,
            -                   err: &errors.errorString{s: "injected status"},
            -           },
            -   },
            +   &{
            +           code: 1,
            +           err: &fmt.wrapError{
            +                   msg: `running Reserve plugin "TestPlugin": injected status`,
            +                   err: &framework.StatusError{message: "injected status"},
            +           },
            +   },
              )
    --- FAIL: TestReservePlugins/UnschedulableAndSuccessReservePlugin (0.00s)
        framework_test.go:2168: Unexpected status (-want, +got):
              (*framework.Status)(
            -   &{
            -           code: 1,
            -           err: &fmt.wrapError{
            -                   msg: `running Reserve plugin "TestPlugin": injected status`,
            -                   err: &errors.errorString{s: "injected status"},
            -           },
            -   },
            +   &{
            +           code: 1,
            +           err: &fmt.wrapError{
            +                   msg: `running Reserve plugin "TestPlugin": injected status`,
            +                   err: &framework.StatusError{message: "injected status"},
            +           },
            +   },
              )
E1206 23:31:29.190208    6469 framework.go:1195] "Failed running Permit plugin" err="injected status" plugin="TestPlugin" pod=""
E1206 23:31:29.190473    6469 framework.go:1195] "Failed running Permit plugin" err="injected status" plugin="TestPlugin" pod=""
--- FAIL: TestPermitPlugins (0.00s)
    --- FAIL: TestPermitPlugins/ErrorPermitPlugin (0.00s)
        framework_test.go:2294: Unexpected status (-want, +got):
              (*framework.Status)(
            -   &{
            -           code: 1,
            -           err: &fmt.wrapError{
            -                   msg: `running Permit plugin "TestPlugin": injected status`,
            -                   err: &errors.errorString{s: "injected status"},
            -           },
            -           failedPlugin: "TestPlugin",
            -   },
            +   &{
            +           code: 1,
            +           err: &fmt.wrapError{
            +                   msg: `running Permit plugin "TestPlugin": injected status`,
            +                   err: &framework.StatusError{message: "injected status"},
            +           },
            +           failedPlugin: "TestPlugin",
            +   },
              )
    --- FAIL: TestPermitPlugins/ErrorAndErrorPlugins (0.00s)
        framework_test.go:2294: Unexpected status (-want, +got):
              (*framework.Status)(
            -   &{
            -           code: 1,
            -           err: &fmt.wrapError{
            -                   msg: `running Permit plugin "TestPlugin": injected status`,
            -                   err: &errors.errorString{s: "injected status"},
            -           },
            -           failedPlugin: "TestPlugin",
            -   },
            +   &{
            +           code: 1,
            +           err: &fmt.wrapError{
            +                   msg: `running Permit plugin "TestPlugin": injected status`,
            +                   err: &framework.StatusError{message: "injected status"},
            +           },
            +           failedPlugin: "TestPlugin",
            +   },
              )
E1206 23:31:29.191668    6469 framework.go:1124] "Failed running Reserve plugin" err="injected status" plugin="test-plugin" pod=""
E1206 23:31:29.191745    6469 framework.go:1033] "Failed running PreBind plugin" err="injected status" plugin="test-plugin" pod="" node=""
E1206 23:31:29.191837    6469 framework.go:1071] "Failed running Bind plugin" err="injected status" plugin="test-plugin" pod="" node=""
E1206 23:31:29.191911    6469 framework.go:1195] "Failed running Permit plugin" err="injected status" plugin="test-plugin" pod=""
E1206 23:31:29.192069    6469 framework.go:1071] "Failed running Bind plugin" err="injected status" plugin="bind-1" pod="" node=""
E1206 23:31:29.192280    6469 framework.go:1071] "Failed running Bind plugin" err="injected status" plugin="bind-0" pod="" node=""
E1206 23:31:29.192527    6469 framework.go:1071] "Failed running Bind plugin" err="injected status" plugin="bind-0" pod="" node=""
FAIL
FAIL    k8s.io/kubernetes/pkg/scheduler/framework/runtime       2.513s```